lib/components/viewers/stop-viewer.js

```diff
@@ -285,20 +285,26 @@ class PatternRow extends Component {
       timeFormat
     } = this.props
     // sort stop times by next departure
-    let sortedStopTimes = null
-    if (stopTimes) {
+    let sortedStopTimes = []
+    const hasStopTimes = stopTimes && stopTimes.length > 0
+    if (hasStopTimes) {
       sortedStopTimes = stopTimes.sort((a, b) => {
         const aTime = a.serviceDay + a.realtimeDeparture
         const bTime = b.serviceDay + b.realtimeDeparture
         return aTime - bTime
       })
       // Cap the number of times shown for any Route at 5. TODO: make configurable
       if (sortedStopTimes.length > 0) sortedStopTimes = sortedStopTimes.slice(0, 5)
-      // Do not show any patterns with no departures happening soon.
-      const timeIsOverThreshold = sortedStopTimes[0].realtimeDeparture - getHomeTime(homeTimezone) > ONE_HOUR_IN_SECONDS * 3
-      if (sortedStopTimes[0] && timeIsOverThreshold) {
-        return null
-      }
+      // Do not show any patterns with no departures happening soon or departures
+      // that have already occurred (there appear to be cases where tomorrow's
```

Comment:
Should the comment say "yesterday's" instead of "tomorrow's"? Also, this seems like a bug that should be reported to the main OTP repo.

Reply:
Let me look into this more... yesterday (Thursday) it was showing times for Friday, so I'm not quite sure what was going on with http://localhost:8080//#/stop/TriMet:13170

Reply:
OK, this query is basically what I was running into yesterday: https://trimet-otp.ibi-transit.com/otp/routers/default/index/stops/TriMet:13170/stoptimes?timeRange=345600&numberOfDepartures=5&startTime=1564686760

This is asking for stop times at 12:12 PM PT on Thursday (with a time range of four days, which is what the param is configured for in the current code). So it makes sense that we would be getting times for Friday. The departure time values in the response are reported in seconds since midnight, so it would make sense that I was seeing these Friday times show up in the UI.

So this doesn't appear to be a bug in OTP, but we should consider whether we want to filter out these times in the request to OTP or here in the UI.

Reply:
I believe prior communication with TriMet revealed that when users are viewing a stop on Friday Monday times should show up for routes that do not have service over the weekend. So we should just leave this as is and remove this filter that I added.